modules/REST/V1/Languages.php

```diff
@@ -627,6 +627,28 @@ public function get_item_schema(): array {
 		return $this->add_additional_fields_schema( $this->schema );
 	}
 
+	/**
+	 * Retrieves an array of endpoint arguments from the item schema for the controller.
+	 * Ensures that the `no_default_cat` property is not returned for `EDITABLE` requests.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $method Optional. HTTP method of the request. The arguments for `CREATABLE` requests are
+	 *                       checked for required values and may fall-back to a given default, this is not done
+	 *                       on `EDITABLE` requests. Default WP_REST_Server::CREATABLE.
+	 * @return array Endpoint arguments.
+	 */
+	public function get_endpoint_args_for_item_schema( $method = WP_REST_Server::CREATABLE ) {
+		if ( WP_REST_Server::EDITABLE === $method ) {
+			$edit_schema = $this->get_item_schema();
+			unset( $edit_schema['properties']['no_default_cat'] );
+
+			return rest_get_endpoint_args_for_schema( $edit_schema, $method );
+		}
+
+		return parent::get_endpoint_args_for_item_schema( $method );
```

Comment:
What do you think of this slightly less verbose version?
Also I would check CREATABLE instead
```suggestion
		$schema = $this->get_item_schema()
		if ( WP_REST_Server::CREATABLE !== $method ) {
			unset( $schema['properties']['no_default_cat'] );
		}
		return rest_get_endpoint_args_for_schema( $schema, $method );
```

Reply:
I'm ok for checking `CREATABLE` instead, that way we ensure covering *all* other methods.
But with your proposal, we may have a problem if `parent::get_endpoint_args_for_item_schema()` changes. We won't benefit from its code anymore, right?

Reply:
I thought to this too but we would have a problem anyway with the direct call to `rest_get_endpoint_args_for_schema()`inside the test.
Hopefully WP will change only the function and never the method.